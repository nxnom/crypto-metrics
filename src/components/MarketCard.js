import React from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';
import { useSelector } from 'react-redux';
import { Link } from 'react-router-dom';
import { Icon } from '@iconify/react';

import { selectCurrency } from 'src/redux/filter/filterSlice';
import { formatCurrency } from 'src/utils/currency';
import styles from './MarketCard.module.css';

function MarketCard({ market, expanded, dark }) {
  const currency = useSelector(selectCurrency);

  return (
    <Link to={`/coins/${market.id}`}>
      <article
        className={cn(
          styles.card,
          expanded && styles.expanded,
          dark && styles.dark,
        )}
      >
        <div
          className={styles.imgContainer}
          style={{ backgroundImage: `url(${market.image})` }}
        >
          {!expanded && (
            <Icon
              className={styles.icon}
              icon="material-symbols:arrow-circle-right-outline"
            />
          )}
        </div>
        <div className={styles.details}>
          <h2 className={styles.name}>{market.name}</h2>
          <p className={styles.price}>
            {formatCurrency(market.price, currency)}
          </p>
        </div>
      </article>
    </Link>
  );
}

MarketCard.propTypes = {
  market: PropTypes.shape({
    id: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    price: PropTypes.number.isRequired,
    image: PropTypes.string.isRequired,
  }).isRequired,
  expanded: PropTypes.bool,
  dark: PropTypes.bool,
};

MarketCard.defaultProps = {
  expanded: false,
  dark: false,
};

export default MarketCard;
