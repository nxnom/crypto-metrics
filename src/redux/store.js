import { configureStore } from '@reduxjs/toolkit';

import api from './services/api';
import filterReducer from './filter/filterSlice';

const store = configureStore({
  reducer: {
    filters: filterReducer,
    [api.reducerPath]: api.reducer,
  },
  middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat(api.middleware),
});

export default store;
