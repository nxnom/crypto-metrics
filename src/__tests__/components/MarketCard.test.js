import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';

import store from 'src/redux/store';
import MarketCard from 'src/components/MarketCard';

describe('MarketCard', () => {
  it('renders correctly', () => {
    const { container } = render(
      <Provider store={store}>
        <MemoryRouter>
          <MarketCard
            market={{
              id: '1',
              name: 'Market 1',
              image: '',
              price: 1,
            }}
          />
        </MemoryRouter>
      </Provider>,
    );
    expect(container).toMatchSnapshot();
  });
});
