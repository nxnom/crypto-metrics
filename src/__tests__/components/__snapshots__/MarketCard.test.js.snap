// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MarketCard renders correctly 1`] = `
<div>
  <a
    href="/coins/1"
  >
    <article
      class="card"
    >
      <div
        class="imgContainer"
        style="background-image: url();"
      >
        <span />
      </div>
      <div
        class="details"
      >
        <h2
          class="name"
        >
          Market 1
        </h2>
        <p
          class="price"
        >
          1.00 $
        </p>
      </div>
    </article>
  </a>
</div>
`;
